test/onnx/test_operators.py

```diff
@@ -249,6 +249,28 @@ def test_conv(self):
         x = torch.ones(20, 16, 50, 40, requires_grad=True)
         self.assertONNX(nn.Conv2d(16, 13, 3, bias=False), x)
 
+    def test_conv_variable_length(self):
+        x = torch.ones(5, 3, 6, 6, requires_grad=True)
+        model = torch.nn.Conv2d(3, 2, 3)
+        y = model(x)
+
+        dynamic_axes = {'input_1': [0, 2, 3], 'output_1': {0: 'output_1_variable_dim_0', 1: 'output_1_variable_dim_1'}}
+        model_proto_name = 'conv2d.onnx'
+        torch.onnx.export(model, x, model_proto_name, verbose=True, input_names=["input_1"], output_names=["output_1"], example_outputs=y, dynamic_axes=dynamic_axes)
```

Comment:
Do you mean when we use dynamic_axes, we must set input_names and output_names as well?

Reply:
Not exactly. User does NOT have to to provide `input_names`/ `output_names` on export API, if they already know what names are being used in the graph, they shall simply provide such names as keys to `dynamic_axes` dictionary and NOT specify `input_names` / `output_names` . however, if they do not know the names OR intend to use different names on exported model, they also have the option. They can do so by providing those names in  `input_names` / `output_names` and using them in `dynamic_axes`.